test/helm/helm_test.go

```diff
@@ -23,38 +29,193 @@ var _ = Describe("Jenkins controller", func() {
 		namespace = e2e.CreateNamespace()
 	})
 	AfterEach(func() {
+		cmd := exec.Command("../../bin/helm", "delete", "jenkins", "--namespace", namespace.Name)
+		output, err := cmd.CombinedOutput()
+		Expect(err).NotTo(HaveOccurred(), string(output))
+
 		e2e.ShowLogsIfTestHasFailed(CurrentGinkgoTestDescription().Failed, namespace.Name)
 		e2e.DestroyNamespace(namespace)
 	})
-	Context("when deploying Helm Chart to cluster", func() {
-		It("creates Jenkins instance and configures it", func() {
-
-			jenkins := &v1alpha2.Jenkins{
-				TypeMeta: v1alpha2.JenkinsTypeMeta(),
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "jenkins",
-					Namespace: namespace.Name,
-				},
-			}
+	It("Deploys Jenkins operator with webhook enabled along with the default jenkins image", func() {
+		jenkins := &v1alpha2.Jenkins{
+			TypeMeta: v1alpha2.JenkinsTypeMeta(),
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "jenkins",
+				Namespace: namespace.Name,
+			},
+		}
+
+		cmd := exec.Command("../../bin/helm", "upgrade", "jenkins", "../../chart/jenkins-operator", "--namespace", namespace.Name, "--debug",
+			"--set-string", fmt.Sprintf("jenkins.namespace=%s", namespace.Name),
+			"--set-string", fmt.Sprintf("operator.image=%s", *imageName), "--install", "--wait")
+		output, err := cmd.CombinedOutput()
+		Expect(err).NotTo(HaveOccurred(), string(output))
+
+		e2e.WaitForJenkinsBaseConfigurationToComplete(jenkins)
+		e2e.WaitForJenkinsUserConfigurationToComplete(jenkins)
+
+	})
 
-			cmd := exec.Command("../../bin/helm", "upgrade", "jenkins", "../../chart/jenkins-operator", "--namespace", namespace.Name, "--debug",
-				"--set-string", fmt.Sprintf("jenkins.namespace=%s", namespace.Name),
-				"--set-string", fmt.Sprintf("operator.image=%s", *imageName), "--install")
-			output, err := cmd.CombinedOutput()
-			Expect(err).NotTo(HaveOccurred(), string(output))
+	It("Deploys Jenkins operator along with webhook and cert-manager", func() {
```

Comment:
Add context like "when deploying the operator and jenkins with security warnings validation enabled". what is being done in the test commands. and later there is 'It' so the desired outcome.

Reply:
Yeah sure,actually I am doing several cases in this it Spec because then I have to redeploy all the resources because of namespace issues